include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7533,6 +7533,13 @@ def err_nsconsumed_attribute_mismatch : Error<
 def err_nsreturns_retained_attribute_mismatch : Error<
   "overriding method has mismatched ns_returns_%select{not_retained|retained}0"
   " attributes">;
+
+def err_nserrordomain_not_tagdecl : Error<
+  "ns_error_domain attribute only valid on enum/struct/union/class">;
```

Comment:
I'm pretty sure it's only valid on an enum. If we wanted to allow it on anything else, it would be typedefs for bare integer types.